funcx_endpoint/tests/smoke_tests/conftest.py

```diff
@@ -0,0 +1,198 @@
+import json
+import os
+
+import pytest
+from globus_sdk import ConfidentialAppAuthClient, AccessTokenAuthorizer
+from funcx import FuncXClient
+from funcx.sdk.executor import FuncXExecutor
+
+# the non-tutorial endpoint will be required, with the following priority order for
+# finding the ID:
+#
+#  1. `--endpoint` opt
+#  2. FUNX_LOCAL_ENDPOINT_ID (seen here)
+#  3. FUNX_LOCAL_ENDPOINT_NAME (the name of a dir in `~/.funcx/`)
+#  4. An endpoint ID found in ~/.funcx/default/endpoint.json
+#
+#  this var starts with the ID env var load
+_LOCAL_ENDPOINT_ID = os.getenv("FUNCX_LOCAL_ENDPOINT_ID")
+
+_CONFIGS = {
+    "prod": {
+        # By default tests are against production, which means we do not need to pass
+        # any arguments to the client object (default will point at prod stack)
+        "client_args": {},
+        # assert versions are as expected on prod
+        "forwarder_version": "0.3.3",
+        "api_version": "0.3.3",
+        "funcx_version": "0.3.3",
+        # This fn is public and searchable
+        "public_hello_fn_uuid": "b0a5d1a0-2b22-4381-b899-ba73321e41e0",
+        # Public tutorial endpoint
+        "tutorial_endpoint": "4b116d3c-1703-4f8f-9f6f-39921e5864df",
+        # other endpoint to test
+        "endpoint_uuid": _LOCAL_ENDPOINT_ID,
+    },
+    "local": {
+        # localhost; typical defaults for a helm deploy
+        "client_args": {
+            "funcx_service_address": "http://localhost:5000/v2",
+            "results_ws_uri": "ws://localhost:6000/ws/v2/",
+        },
+        "endpoint_uuid": _LOCAL_ENDPOINT_ID,
+    },
+}
+
+
+def _get_local_endpoint_id():
+    # get the ID of a local endpoint, by name
+    # this is only called if
+    #  - there is no endpoint in the config (e.g. config via env var)
+    #  - `--endpoint` is not passed
+    local_endpoint_name = os.getenv("FUNCX_LOCAL_ENDPOINT_NAME", "default")
+    data_path = os.path.join(
+        os.path.expanduser("~"), ".funcx", local_endpoint_name, "endpoint.json"
+    )
+
+    try:
+        with open(data_path) as fp:
+            data = json.load(fp)
+    except Exception:
+        return None
```

Comment:
That exception is raised in the github actions test environment because a local endpoint doesn't exist. A local endpoint is not required at all right now because the smoke tests run against the tutorial_endpoint alone. I think the proper fix here would be to not have a redundant `tutorial_endpoint` and `endpoint_uuid` option. Let me refactor this a bit.

Reply:
Okay. When I was tinkering with the tests, I thought that having support for testing with both a local EP and the tutorial EP in a single run was intentional, so I was careful to preserve that.

If we don't care about that case (I don't have a reason we would; I just assumed it was supported on purpose), maybe we just make it the `endpoint_id` field, and it defaults to the prod tutorial EP? I'll trust your judgement on this.

Reply:
Right, we don't need another field here really. It's just that smoke_tests assume that the target EP is the tutorial ep to make sure that it is up, so it's simpler to make that a default.